### PR TITLE
add review comment to sb files

### PR DESCRIPTION
**Pull request by oleksandr-didyk**
### Context

Contributes to https://github.com/dotnet/source-build/issues/3435

Adds comments to source-build files asking for the inclusion of the source-build team in PRs that alter `SourceBuild*` files. Non-reviewed changes could potentially cause issues down the line, be it in the downstream repos or the product build (as has happened in the past, see https://github.com/dotnet/source-build/issues/3435#issuecomment-1570650046)

### Changes Made

  - added comments to source-build files asking for the inclusion of the source-build team in PRs that alter `SourceBuild*` file.

### CODE REVIEW

**Review by JanKrivanek — APPROVED** [review]

**Review by ladipro — APPROVED** [review]
Curious if we shouldn't add the `CODEOWNERS` file to enforce such rules.

**Comment by oleksandr-didyk** [comment]
> Curious if we shouldn't add the `CODEOWNERS` file to enforce such rules.

From past experience `CODEOWNERS` is a useful tool, but if you are tagging a team alias it will need to have write access to the repo for the entry to work.

Speaking of write access - I don't have it for `msbuild` so I cannot merge this PR myself. Would be great if someone from the maintainers could merge it. Thanks!

**Pull request merged**
